src/components/home/index.js

```diff
@@ -5,25 +5,45 @@ import { Controller, useForm } from "react-hook-form";
 import { Button, TextField, Typography } from "@mui/material";
 import * as authService from "../../services/auth-service";
 import * as tuitService from "../../services/tuits-service";
-
+import { useNavigate } from "react-router-dom";
+import { useSearchParams } from 'react-router-dom'
 const Home = () => {
+
+  const navigate = useNavigate();
   const [authprofile, setAuthProfile] = useState({});
   const [tuits, setTuits] = useState();
-
+  const [searchParams, setSearchParams] = useSearchParams();
+  let searchQuery=searchParams.get('q');
   useEffect(() => {
     const fetchProfile = async () => {
       try {
         const user = await authService.profile();
         setAuthProfile(user);
         const alltuit = await tuitService.findAllTuits();
+        
+        const tempTuits=[];
+if(searchQuery!==null && searchQuery.length>0 ){
+  searchQuery=searchQuery.toUpperCase();
+        for(let i=0;i<alltuit.length;i++){
+
+          const tuit=alltuit[i].tuit.toUpperCase();
+
+          if(tuit.indexOf(searchQuery)!==-1){
+            tempTuits.push(alltuit[i]);
+          }
+        }
+        setTuits(tempTuits);
+      } else {
         setTuits(alltuit);
+      }
+        
       } catch (e) {
-        // console.log("session profile not found, send to login page");
-        // navigate("/profile/login");
+         console.log("session profile not found, send to login page");
+         navigate("/profile/login");
```

Comment:
I do not agree with this function. If the authProfile is not available. All tuits should still be displayed so that you can view tuits without having to login. I created a small change that I updated to Main last night that would display this feature. But you had already create your integration 5 branch before that small changed was push to main. I'm going to make a merge from main to your integration 5 with the small change. Then you need to pull from your integration 5 to propagate the change to all your other branches. There might be a merge conflict if you push this to integration-5 after main is push into integration-5. 

I think the search feature is good. Some odd implementation since search component can only work on home page and not any other page even though the search component is on every page. Since it's an extra feature I am ok with it.

Reply:
Done!